### ISSUE TITLE
GetOperationStatus has no retries for 502 status code

### ISSUE THREAD

**Issue by igsaf2** [databricks/databricks-sql-python#218]
we use dbt-databricks (which uses databricks-sql-connector)

databricks-sql-connector==2.6.2 (but seems to be relevant to current version of the library.
dbt-databricks==1.5.4

And we experience lots of errors if our databricks cluster is under heavy load. And restarts cause more load...

Relevant traceback:
```
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: <class 'databricks.sql.exc.RequestError'>: Error during request to server
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: attempt: 1/30
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: bounded-retry-delay: None
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: elapsed-seconds: 6.919969797134399/900.0
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: error-message:
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: http-code: 502
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: method: GetOperationStatus
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: no-retry-reason: non-retryable error
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: original-exception:
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: query-id: b'\xac\x18\xa1\xe2\x1a\x03@S\xbf\x1a\xca\x9ad(W\x8e'
[2023-09-07, 11:04:25 UTC] {subprocess.py:93} INFO - [0m11:04:25  Databricks adapter: session-id: None
```

**Comment by susodapop** [comment]
This is exactly the type of issue that https://github.com/databricks/databricks-sql-python/pull/182 addressed. Historically 502 codes are never retried because they cannot be guaranteed to be idempotent. Under the new retry policy (which is available in connector version >=2.9.3) you can _optionally_ specify codes that should be retried beyond 429 and 503. 

dbt-databricks==1.6.3 is the first version of dbt-databricks that installs databricks-sql-connector==2.9.3. With this version installed you can pass the retry configuration as part of your `profiles.yml` like this:

```yaml
your_project_name:
  target: dev
  outputs:
    dev:
      threads: 4
      connection_parameters:
        _enable_v3_retries: true
        _retry_dangerous_codes:
          - 500
```

**Comment by susodapop** [comment]
Also, there's an example of configuring the retry policy here https://github.com/databricks/databricks-sql-python/blob/main/examples/v3_retries_query_execute.py
